lib/grpc/client/adapters/mint.ex

```diff
@@ -15,7 +15,14 @@ defmodule GRPC.Client.Adapters.Mint do
 
   @impl true
   def connect(%{host: host, port: port} = channel, opts \\ []) do
-    opts = Keyword.merge(@default_connect_opts, connect_opts(channel, opts))
+    # Added :config_options to facilitate testing.
+    {config_opts, opts} = Keyword.pop(opts, :config_options, [])
+    # Merges default opts with application specific opts set
```

Comment:
I think this comment is not necessary. The code is self explanatory

Reply:
This comment about how pass the options to the adapter would make more sense to be in the documentation via README or something like that

Reply:
Good call. Added suggested changes.